File: src/tools/eraser.js
import { PencilTool } from './pencil.js';

export class EraserTool extends PencilTool {
    constructor(app) {
        super(app);
        this.name = 'eraser';
    }

    onMouseDown(e) {
        super.onMouseDown(e);
        if (this.currentPath) {
            this.currentPath.strokeColor = '#000000'; // Placeholder
            this.currentPath.composite = 'destination-out';
            // Make eraser 4x thicker than current stroke, or at least 10px
            const currentWidth = this.app.tools.style.strokeWidth || 2;
            this.currentPath.strokeWidth = Math.max(currentWidth * 4, 10);
        }
    }
}

File: wasm/tools/eraser.go
package tools

// EraserTool is a pencil stroke rendered with destination-out blending
// (port of eraser.js).
type EraserTool struct {
	PencilTool
}

func NewEraserTool(env *Env) *EraserTool {
	t := &EraserTool{PencilTool: PencilTool{BaseTool: newBase(env, "eraser")}}
	return t
}

func (t *EraserTool) OnPointerDown(e *PointerEvent) {
	t.PencilTool.OnPointerDown(e)
	if t.current != nil {
		t.current.Stroke = "#000000" // placeholder, erased by blend mode
		t.current.Composite = "destination-out"
		// Eraser is 4x the current stroke, at least 10px
		w := t.env.Style.StrokeWidth
		if w <= 0 {
			w = 2
		}
		if w*4 > 10 {
			t.current.StrokeWidth = w * 4
		} else {
			t.current.StrokeWidth = 10
		}
	}
}
